sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVFileFormat.scala

```diff
@@ -125,7 +125,11 @@ class CSVFileFormat extends TextBasedFileFormat with DataSourceRegister {
         actualRequiredSchema,
         parsedOptions,
         actualFilters)
-      val schema = if (isColumnPruningEnabled) actualRequiredSchema else actualDataSchema
+      // Use column pruning when specified by Catalyst, except when one or more columns have
+      // existence default value(s), since in that case we instruct the CSV parser to disable column
+      // pruning and instead read each entire row in order to correctly assign the default value(s).
+      val useColumnPruningForCheckingHeader = isColumnPruningEnabled
+      val schema = if (useColumnPruningForCheckingHeader) actualRequiredSchema else actualDataSchema
```

Comment:
Let's inline `useColumnPruningForCheckingHeader`:
```suggestion
      val schema = if (isColumnPruningEnabled) actualRequiredSchema else actualDataSchema
```

Reply:
Sounds good, done.